test/banach/retry_test.clj

```diff
@@ -5,6 +5,40 @@
             [clojure.math.numeric-tower :as math]
             [banach.retry :as retry]))
 
+(def ctx-1 {:failures [(Exception. "Slow your roll")]})
+(def ctx-2 (update ctx-1 :failures conj (Exception. "KABOOM")))
+
+(deftest routing-tests
+  (let [strat (retry/routing
+               (comp odd? count :failures) identity
+               (constantly true) retry/give-up)]
```

Comment:
I had a bit of trouble understanding this strategy and I'd like to make sure that I am:

1. `(comp odd? count :failures) identity` - if the `context` contains an odd number of failures, return the identity of the `context`, which is the `context` itself. If this fails, then fall through to the next route.
2. `(constantly true) retry/give-up)` if this has been reached, throw the most recently captured exception.